apps/kol-amelamdim/middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { verify } from './services/jwt_sign_verify';
+
+export async function middleware(request: NextRequest) {
+  const { pathname } = request.nextUrl;
+  const userToken = request.cookies.get('token');
+  if (request.nextUrl.pathname.startsWith('/admin/')) {
+    const jwt = request.cookies.get('adminToken');
+    if (jwt === undefined) {
+      request.nextUrl.pathname = '/login';
+      return NextResponse.redirect(request.nextUrl);
+    }
+    try {
+      verify(jwt, process.env.ADMIN_TOKEN_SECRET);
+      return NextResponse.next();
+    } catch (error) {
+      request.nextUrl.pathname = '/login';
+      return NextResponse.redirect(request.nextUrl);
+    }
+  } else if (pathname.startsWith('/api/upload-file')) {
```

Comment:
I think about that and we should not "protect" a specific route every time. Instead we should "group" the routes that we want to protect for example all the routes that start with admin or let׳s say all the routes that start with /api/files it will be overkill to manage that like this